### PR TITLE
bankrupt the bucket when there is LP but no deposit/collateral

### PR DESCRIPTION
**Pull request by EdNoepel**
@mattcushman Could you craft a test that reproduces this in a bucket with no deposit, but 1 wei of collateral?  Likely need to cover that case as well.

Implemented `BucketBankruptcy` event (#496), which now needs unit tests for liquidations.

### CODE REVIEW

**Review by grandizzy — COMMENTED** [review]
lgtm, did a PR to read only once from Bucket storage struct, bankruptBucketIfEmpty inline (can do function if needed for move or even keep logic inline there too)  https://github.com/ajna-finance/contracts/pull/498

**Comment by grandizzy** [comment]
> lgtm, did a PR to read only once from Bucket storage struct, bankruptBucketIfEmpty inline (can do function if needed for move or even keep logic inline there too) #498

oops, hit stack too deep when reading LPs only once, will make a fix for

**Comment by EdNoepel** [comment]
@grandizzy 's changes have been merged.  @mattcushman , please review for inclusion into your branch.

File: src/libraries/external/LenderActions.sol [hunk 1]
```diff
@@ -621,9 +639,10 @@ library LenderActions {
...
-        removedAmount_ = Maths.wmul(depositScale, unscaledRemovedAmount);        
-        // calculate scale amount remaining
-        uint256 remaining = Maths.wmul(depositScale, unscaledDepositAvailable - unscaledRemovedAmount);
+        removedAmount_     = Maths.wmul(depositScale, unscaledRemovedAmount);        
+        // calculate amount remaining
+        unscaledRemaining_ = unscaledDepositAvailable - unscaledRemovedAmount;
+        uint256 remaining  = Maths.wmul(depositScale, unscaledRemaining_);
 
         // abandon dust amounts upon last withdrawal
         if (remaining < params_.dustLimit && redeemedLPs_ == params_.bucketLPs) {
```
Comment (mattcushman) [inline src/libraries/external/LenderActions.sol:648]: Do we need to also set _unscaledRemaining to 0 here?  It looks to me that it should.  It gets pass out of the function and used in the check to see if there should be a bucketBankruptcy called.

Comment (EdNoepel) [reply src/libraries/external/LenderActions.sol:648]: Yes; good catch.  Will do today.

Comment (EdNoepel) [reply src/libraries/external/LenderActions.sol:648]: fixed in b93bb94ab35728ab6399b1b629fb81810a85eba9

**Review by mattcushman — APPROVED** [review]
lgtm

**Pull request merged**